### PR TITLE
(#3786) Add feature card analytics

### LINKED ISSUE #3786: Implement analytics for NCIDS feature cards
### Issue description
Implement the analytics to be captured for NCIDS feature cards and rows.

> **ESTIMATE** TBD

### What's the expected change?
- Analytics are passed for clicks/use of the NCIDS feature cards and rows.

### What's the current functionality?
- Analytics are not passed for NCIDS feature cards and rows.
- 
### What's the updated acceptance criteria?
```gherkin
```

### Related Tickets
* Feature card row: #3492 
* Feature cards: #3611, #3612

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #3786

https://ncigovcdode558.prod.acquia-sites.com/ncids-internal-feature-card-test
https://ncigovcdode558.prod.acquia-sites.com/ncids-external-feature-card-test

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/core/analytics/landing-page-contents-helper.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * feature cards, guide cards, raw html blocks, etc. The data-eddl attribute
+ * is required on all elements that should be counted as a row.
+ *
+ * @example
+ * const rowsTotal = getPageRows().length;
+ *
+ * @returns {HTMLElement[]} Array of HTML element "rows" found in the document.
+ */
+export const getPageRows = (): HTMLElement[] => {
+	const rows = document.querySelectorAll('[data-eddl]');
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/core/analytics/landing-page-contents-helper.ts:44]: This should be more specific, as we have other `data-eddl-*` data attributes that are not home/landing.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
+import { trackOther } from '../../core/analytics/eddl-util';
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts:1]: This should be a different function that wraps trackOther... e.g.

```
  const trackLandingClick = (row: number, column: number, other) => {
   if missing stuff error
 trackOther({
     common hard coded stuff,
     row,
     blah
  })
}
```

Comment (bryanpizzillo) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts:1]: This is the common function...

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+ *     the user clicks on the secondary link, pass the text displayed for that
+ *     link. If the link area cannot be determined use "Undefined".
+ * @param linkArea - Location that was clicked. "Image", "Description",
+ * 			"Title", or "Secondary Link". If the link area cannot be determined use
+ * 			"Undefined".
+ * @param totalLinks - Total number of links shown in the card.
+ * @param linkPosition - The index of the link clicked in the card evaluated by
+ * 			counting the links top to bottom.
+ */
+const track = (
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts:33]: As I look below this looks more like the common helper function.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-3-feature-card-row.html.twig [hunk 4]
```diff
@@ -1,15 +1,15 @@
 {% if (content.field_row_cards['#items']|length > 0) %}
   {% if content.field_container_heading[0] %}
-    <section class="usa-section usa-section--light cgdp-feature-card-row">
-      <div class="grid-container">     
+    <section class="usa-section usa-section--light cgdp-feature-card-row" data-eddl="feature-card">
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-3-feature-card-row.html.twig:3]: When I said `data-eddl-something` I did not mean that everything should be `data-eddl`. This is like having `<div data="foo">` -- kind of makes it hard to use it for multiple purposes.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
It is getting there, but I think there is a bit more in the TS that you can make common.

**Comment by bennettcc** [comment]
Approving on behalf of analytics team - Joe/Allison reviewed and approved on 5/23

**Review by dlescarbeau — APPROVED** [review]

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
I got some requests...

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+/**
+ * Gets text of area of card user clicked.
+ * @param {Event} evt - Click event
+ */
+const getLinkText = (evt: Event): string => {
+	const anchorEl = evt.currentTarget as HTMLElement;
+	const linkArea = getLinkArea(evt);
+
+	const textOptions: { [key: string]: string } = {
+		Image: 'Image',
```
[the remark this answers is not in the document]
Comment (olitharp-nci) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-feature-card-row/cgdp-feature-card-row.ts]:
```
  | linkText | This value will change slightly based on the linkArea. If a user clicks on the title, this should pass the text of the title value (truncate after 50). If the user clicks on the description, this should pass the text of the description value (truncate after 50). If the user clicks on the image, pass “Image”. If the user clicks on the secondary link, pass the text displayed for that link. |
  | linkArea | For feature cards, this value should be set to “Image”, “Description”, “Title”, or “Secondary Link” based on where the user clicks. If the area cannot be determined, please pass "Undefined" |
 ```

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/entrypoints/home-landing/home-landing.ts [hunk 6]
```diff
@@ -4,6 +4,8 @@ import carousel from 'Core/libraries/carousel/carousel';
 import './home-landing.scss';
 import './home-landing-legacy.scss';
 
+import cgdpFeatureCardRow from '../../lib/components/cgdp-feature-card-row';
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/entrypoints/home-landing/home-landing.ts:7]: This is not correct, should be `ncids-home-landing.ts`

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**